### ISSUE TITLE
ImportError: No module named ndg.httpsclient.ssl_peer_verification

### ISSUE THREAD

**Issue by MadSpindel** [pusher/pusher-http-python#65]
I get the following on a Ubuntu server:

    pusher = Pusher(app_id=settings.PUSHER_APP_ID, key=settings.PUSHER_KEY, secret=settings.PUSHER_SECRET)
      File "/usr/local/lib/python2.7/dist-packages/pusher/pusher.py", line 42, in __init__
    from pusher.requests import RequestsBackend
      File "/usr/local/lib/python2.7/dist-packages/pusher/requests.py", line 12, in <module>
    import urllib3.contrib.pyopenssl
      File "/usr/lib/python2.7/dist-packages/urllib3/contrib/pyopenssl.py", line 23, in <module>
    from ndg.httpsclient.ssl_peer_verification import SUBJ_ALT_NAME_SUPPORT
    ImportError: No module named ndg.httpsclient.ssl_peer_verification

I installed pusher with pip install pusher

**Comment by MadSpindel** [comment]
Updating pyOpenSSL and/or urllib3 and rebooting the server fixed the error.

**Comment by MatthiasLohr** [comment]
Not for me. I still have this issue, which versions do yo use?

**Comment by MadSpindel** [comment]
I use:
```
pusher==1.2.3
pyOpenSSL==0.15.1
urllib3==1.11
```

I don't remember if I did something more to fix this error.

**Comment by james-hoegerl** [comment]
Any luck @MatthiasLohr? Struggling with this in a deployment package for AWS Lambda.

**Comment by MatthiasLohr** [comment]
@james-hoegerl, not for this special case, sorry.

But i solved a similar problem with the requests module by downgrading requests to a lower version. More a workaround than a solution, i know... :(

**Comment by rafaelafonso** [comment]
@MatthiasLohr I have found the same error (trying to make it work with GAE) and I fixed it creating an empty __init__.py on /ndg folder. Maybe you want to give it a try ;)

**Comment by james-hoegerl** [comment]
Wow you just made my month.

Ran into this now though.

```
libssl.so.1.0.0: cannot open shared object file: No such file or directory: ImportError
Traceback (most recent call last):
  File "/var/task/Lambda.py", line 3, in pusherTest
    mypusher = Pusher(app_id=u'154504', key=u'121300e92c61a1fe43bc', secret=u'bfff467a2b6551f19226')
  File "/var/task/pusher/pusher.py", line 42, in __init__
    from pusher.requests import RequestsBackend
  File "/var/task/pusher/requests.py", line 12, in <module>
    import urllib3.contrib.pyopenssl
  File "/var/task/urllib3/contrib/pyopenssl.py", line 54, in <module>
    import OpenSSL.SSL
  File "/var/task/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import rand, crypto, SSL
  File "/var/task/OpenSSL/rand.py", line 12, in <module>
    from OpenSSL._util import (
  File "/var/task/OpenSSL/_util.py", line 6, in <module>
    from cryptography.hazmat.bindings.openssl.binding import Binding
  File "/var/task/cryptography/hazmat/bindings/openssl/binding.py", line 15, in <module>
    from cryptography.hazmat.bindings._openssl import ffi, lib
ImportError: libssl.so.1.0.0: cannot open shared object file: No such file or directory
```

**Comment by MatthiasLohr** [comment]
@james-hoegerl you have to install the libssl package from your distribution.

**Comment by james-hoegerl** [comment]
Thanks for the help. Sorry I'm a bit new. That's not a python package but a C library? If it's not installed on the Lambda service it seems I may be out of luck.

**Comment by ABewsher** [comment]
Hi.

I had the initial issue and adding the __init__.py file fixed that problem.  I then encountered the second problem where I could not open the shared object files.  The issue then was that I am developing on a Mac, but the Lambda is going to be running on AWS running Amazon Linux.  I needed to log into my EC2 instance and do pip install ... from there.  Then copy the files from there to my local machine to package them up and upload them to the AWS Lambda.

However... after getting that far, I have yet another issue!  Now I get:

`No module named cryptography.hazmat.bindings.openssl.binding: ImportError Traceback (most recent call last): File "/var/task/main.py", line 28, in handler pusher = Pusher(app_id=u'161101', key=u'072caa5b47bebceba581', secret=u'7facde1911e94efa1ef8', port=443, ssl=True) File "/var/task/pusher/pusher.py", line 42, in __init__ from pusher.requests import RequestsBackend File "/var/task/pusher/requests.py", line 12, in <module> import urllib3.contrib.pyopenssl File "/var/task/urllib3/contrib/pyopenssl.py", line 54, in <module> import OpenSSL.SSL File "/var/task/OpenSSL/__init__.py", line 8, in <module> from OpenSSL import rand, crypto, SSL File "/var/task/OpenSSL/rand.py", line 12, in <module> from OpenSSL._util import ( File "/var/task/OpenSSL/_util.py", line 6, in <module> from cryptography.hazmat.bindings.openssl.binding import Binding ImportError: No module named cryptography.hazmat.bindings.openssl.binding`

I've run the following commands in my virtualenv:

sudo yum -y install libffi-devel
pip install pusher
pip install cryptography

I've also created the __init__.py file in the ndg directory.

But when I upload all the files, I get the above error.

**Comment by james-hoegerl** [comment]
Man I had that error at one point and I can't remember what fixed it. I did get mine running by using an Amazon Linux instance to create the distro. Man lemme look back at mine and see. I wanna say you need to yum install another package. Or make sure when you do install the cryptography package nothing is failing. I'll check and get back to you.

**Comment by ABewsher** [comment]
Woot!  Fixed it.

When I'm making the zip, I needed to include the contents of the dist-packages directories from both _lib_ and _lib64_ !

Now it's not complaining about modules .. just bugs in my code :-P

Thanks for your help.  The steps in this thread were the answer.

I also got inspiration by looking in the scripts in this repository https://github.com/snowflakedb/examples/tree/master/lambda_function
